spec/params/params.spec.ts

```diff
@@ -0,0 +1,212 @@
+import { expect } from "chai";
+import * as params from "../../src/params";
+
+describe("Params spec extraction", () => {
+  it("converts Expressions in the param default to strings", () => {
+    const bar = params.defineInt("BAR");
+    expect(
+      params.defineString("FOO", { default: bar.notEquals(22).then("asdf", "jkl;") }).toSpec()
+        .default
+    ).to.equal(`{{ params.BAR != 22 ? "asdf" : "jkl;" }}`);
+  });
+});
+
+describe("Params value extraction", () => {
+  beforeEach(() => {
+    process.env.A_STRING = "asdf";
+    process.env.SAME_STRING = "asdf";
+    process.env.DIFF_STRING = "jkl;";
+    process.env.AN_INT = "-11";
+    process.env.SAME_INT = "-11";
+    process.env.DIFF_INT = "22";
+    process.env.PI = "3.14159";
+    process.env.TRUE = "true";
+    process.env.FALSE = "false";
+  });
+
+  afterEach(() => {
+    params.clearParams();
+    delete process.env.A_STRING;
+    delete process.env.SAME_STRING;
+    delete process.env.DIFF_STRING;
+    delete process.env.AN_INT;
+    delete process.env.SAME_INT;
+    delete process.env.DIFF_INT;
+    delete process.env.TRUE;
+    delete process.env.PI;
+    delete process.env.TRUE;
+    delete process.env.FALSE;
+  });
+
+  it("extracts identity params from the environment", () => {
+    const strParam = params.defineString("A_STRING");
+    expect(strParam.value()).to.equal("asdf");
+
+    const intParam = params.defineInt("AN_INT");
+    expect(intParam.value()).to.equal(-11);
+
+    const boolParam = params.defineBoolean("TRUE");
+    expect(boolParam.value()).to.be.true;
+
+    const floatParam = params.defineFloat("PI");
+    expect(floatParam.value()).to.equal(3.14159);
+
+    const falseParam = params.defineBoolean("FALSE");
+    expect(falseParam.value()).to.be.false;
+  });
+
+  it("falls back on the javascript zero values in case of type mismatch", () => {
```

Comment:
hmm fallbacks are tricky. We don't expect these to happen in real life no? (b/c cli will ensure that the values exist)

Reply:
Correct.